geonode/base/api/filters.py

```diff
@@ -51,12 +51,12 @@ class FavoriteFilter(BaseFilterBackend):
 
     def filter_queryset(self, request, queryset, _):
         if strtobool(request.query_params.get("favorite", 'False')):
-            ctype = list({r.resource_type for r in queryset})
+            c_types = list({r.polymorphic_ctype.model for r in queryset})
```

Comment:
@marthamareal what's the difference between `r.resource_type` and `r.polymorphic_ctype.model`?

Reply:
For dashboard and geostories, their resource_type is assigned as `dashboard, geostory`, however ctype_model  in Favorite relation is `geoapp.`  hence if we consider resourcetypes on filtering Favorites([here](https://github.com/GeoNode/geonode/pull/9390/files#diff-ae772dfe9ea2394b6260657dfbc90bb4c123b09096a60959b05e889c554a76ceR59)), geoapps will be missed